Sources/Client/Client/ClientLogger.swift

```diff
@@ -147,9 +147,35 @@ public final class ClientLogger {
     /// - Parameters:
     ///     - icon: a small icon string like a tag for messages, e.g. 🦄
     ///     - dateAndTime: a formatted string of date and time, could be empty.
+    ///     - message: a message.
+    @available(*, deprecated, message: """
+    Customizing the logger block with `ClientLogger.logger` is deprecated. Please use `ClientLogger.log` to customize your logging logic.
+    """)
+    public static var logger: (String, String, String) -> Void = { icon, dateTime, message in
+        if iconEnabled {
+            print(icon, terminator: " ")
+        }
+
+        if dateEnabled {
+            if dateTime.isEmpty {
+                print("[\(Date().log)]", terminator: " ")
+            } else {
+                print("[\(dateTime)]", terminator: " ")
+            }
+        }
+
+        print(message)
+    }
+    
+    /// A customizable log block.
+    /// By default error messages will print to the console, but you can customize it to use your own log block.
+    ///
+    /// - Parameters:
+    ///     - icon: a small icon string like a tag for messages, e.g. 🦄
+    ///     - dateAndTime: a formatted string of date and time, could be empty.
     ///     - level: Log level
     ///     - message: a message.
-    public static var logger: (String, String, Level, String) -> Void = { icon, dateTime, level, message in
+    public static var log: (String, String, Level, String) -> Void = { icon, dateTime, level, message in
```

Comment:
To prevent future breaking changes, what if we wrap these values into something like:
```
struct LogItem {
  let message: String
  let dateTime: Date ....
}
```
It would allow as adding and changing log-related things much more easily.

Reply:
We'll break it one more time in v3, I suggest we do that then.
I don't want to improve something I'm sure we'll replace soon

Reply:
Using `LogItem` would open a possibility to not have to break it in v3 again. It's up to you 🤷‍♂️

Reply:
I really like the LogItem solution so I'll definitely use it in the new implementation :D But even if I've used it now, I'm sure there'll still be breaking changes (so no breaking changes is not a possibility)
So I suggest we keep it as it is (more familiar with the old approach) and break it in v3 for a much better logger